File: examples/clients/WorkflowClient.cs
/// <summary>
// Console.WriteLine($"Status: {result.Status}, Duration: {result.Duration}");
// var result = await client.WaitForCompletionAsync(execution.ExecutionId);
// // Wait for completion
// 
// Console.WriteLine($"Execution started: {execution.ExecutionId}");
// 
// });
//     ["destination"] = "database"
//     ["source"] = "api",
// {
// var execution = await client.ExecuteWorkflowAsync("data-sync", new Dictionary<string, object?>
// // Async execution
// 
// var client = new WorkflowClient("https://workflow-engine.example.com", apiKey: "your-api-key");
// Usage example:

}
    }
        throw new OperationCanceledException();
        
        }
            await Task.Delay(pollInterval.Value, cancellationToken);
            
                return status;
            if (status.Status is ExecutionStatus.Completed or ExecutionStatus.Failed or ExecutionStatus.Cancelled)
            
            var status = await GetExecutionStatusAsync(executionId, cancellationToken);
        {
        while (!cancellationToken.IsCancellationRequested)
        
        pollInterval ??= TimeSpan.FromSeconds(2);
    {
        CancellationToken cancellationToken = default)
        TimeSpan? pollInterval = null,
        Guid executionId,
    public async Task<ExecutionStatusDetail> WaitForCompletionAsync(
    /// </summary>
    /// Wait for execution to complete 🎯
    /// <summary>
    
    }
        return await response.Content.ReadFromJsonAsync<ExecutionStatusDetail>(cancellationToken: cancellationToken);
        response.EnsureSuccessStatusCode();
        
            cancellationToken);
            $"{_baseUrl}/api/v1/workflows/executions/{executionId}",
        var response = await _httpClient.GetAsync(
    {
        CancellationToken cancellationToken = default)
        Guid executionId,
    public async Task<ExecutionStatusDetail> GetExecutionStatusAsync(
    /// </summary>
    /// Get execution status 📊
    /// <summary>
    
    }
        return await response.Content.ReadFromJsonAsync<ExecutionResponse>(cancellationToken: cancellationToken);
        response.EnsureSuccessStatusCode();
        
            cancellationToken);
            request,
            $"{_baseUrl}/api/v1/workflows/execute/{workflowName}",
        var response = await _httpClient.PostAsJsonAsync(
        
        };
            TimeoutSeconds = timeoutSeconds
            WaitForCompletion = true,
            Inputs = inputs,
        { 
        var request = new ExecuteWorkflowRequest 
    {
        CancellationToken cancellationToken = default)
        int timeoutSeconds = 300,
        Dictionary<string, object?>? inputs = null,
        string workflowName,
    public async Task<ExecutionResponse> ExecuteWorkflowSyncAsync(
    /// </summary>
    /// Execute and wait for completion (synchronous) ⏳
    /// <summary>
    
    }
        return await response.Content.ReadFromJsonAsync<ExecutionResponse>(cancellationToken: cancellationToken);
        response.EnsureSuccessStatusCode();
        
            cancellationToken);
            request,
            $"{_baseUrl}/api/v1/workflows/execute/{workflowName}",
        var response = await _httpClient.PostAsJsonAsync(
        var request = new ExecuteWorkflowRequest { Inputs = inputs };
    {
        CancellationToken cancellationToken = default)
        Dictionary<string, object?>? inputs = null,
        string workflowName,
    public async Task<ExecutionResponse> ExecuteWorkflowAsync(
    /// </summary>
    /// Execute a workflow and get the execution ID ✨
    /// <summary>
    
    }
            _httpClient.DefaultRequestHeaders.Add("X-API-Key", apiKey);
        if (apiKey != null)
        _httpClient = new HttpClient();
        _baseUrl = baseUrl;
    {
    public WorkflowClient(string baseUrl, string? apiKey = null)
    
    private readonly string _baseUrl;
    private readonly HttpClient _httpClient;
{
public class WorkflowClient
/// </summary>
/// 💎 Official C# client SDK for the workflow engine

File: examples/clients/WorkflowClient.py
# 🐍 Python client SDK

import requests
from typing import Dict, Any, Optional
import time

class WorkflowClient:
    def __init__(self, base_url: str, api_key: Optional[str] = None):
        self.base_url = base_url
        self.api_key = api_key
        self.session = requests.Session()
        if api_key:
            self.session.headers['X-API-Key'] = api_key
    
    def execute_workflow(
        self, 
        workflow_name: str, 
        inputs: Optional[Dict[str, Any]] = None
    ) -> Dict[str, Any]:
        """Execute a workflow ✨"""
        response = self.session.post(
            f"{self.base_url}/api/v1/workflows/execute/{workflow_name}",
            json={"inputs": inputs or {}}
        )
        response.raise_for_status()
        return response.json()
    
    def execute_workflow_sync(
        self,
        workflow_name: str,
        inputs: Optional[Dict[str, Any]] = None,
        timeout_seconds: int = 300
    ) -> Dict[str, Any]:
        """Execute and wait for completion ⏳"""
        response = self.session.post(
            f"{self.base_url}/api/v1/workflows/execute/{workflow_name}",
            json={
                "inputs": inputs or {},
                "waitForCompletion": True,
                "timeoutSeconds": timeout_seconds
            }
        )
        response.raise_for_status()
        return response.json()
    
    def get_execution_status(self, execution_id: str) -> Dict[str, Any]:
        """Get execution status 📊"""
        response = self.session.get(
            f"{self.base_url}/api/v1/workflows/executions/{execution_id}"
        )
        response.raise_for_status()
        return response.json()
    
    def wait_for_completion(
        self,
        execution_id: str,
        poll_interval: float = 2.0,
        timeout: Optional[float] = None
    ) -> Dict[str, Any]:
        """Wait for execution to complete 🎯"""
        start_time = time.time()
        
        while True:
            status = self.get_execution_status(execution_id)
            
            if status['status'] in ['Completed', 'Failed', 'Cancelled']:
                return status
            
            if timeout and (time.time() - start_time) > timeout:
                raise TimeoutError(f"Execution timed out after {timeout} seconds")
            
            time.sleep(poll_interval)

# Usage:
# client = WorkflowClient('https://workflow-engine.example.com', api_key='your-api-key')
# 
# # Execute workflow
# execution = client.execute_workflow('data-sync', {
#     'source': 'api',
#     'destination': 'database'
# })
# 
# print(f"Execution started: {execution['executionId']}")
# 
# # Wait for completion
# result = client.wait_for_completion(execution['executionId'])
# print(f"Status: {result['status']}, Duration: {result['duration']}")

